docs/book/validators/phone-number.md

```diff
@@ -0,0 +1,143 @@
+    'other-data' => '…',
+];
+
+$validator->isValid($postPayload['number'], $postPayload); // true
+```
+
+## Limiting Acceptable Number Types
+
+The validator is capable of validating a range of number types; mobile numbers, fixed lines, premium rate and emergency numbers to name a few.
+By default, the validator will accept any number type:
+
+```php
+$validator = new PhoneNumber([
+    'country' => 'US',
+]);
+$validator->isValid('911'); // true
+```
+
+In order to reduce the range of acceptable number types, provide the validator with a bitmask of types from `PhoneNumberValue` value object:
+
+```php
+use Laminas\I18n\PhoneNumber\PhoneNumberValue;
+use Laminas\I18n\PhoneNumber\Validator\PhoneNumber;
+
+$allowed = PhoneNumberValue::TYPE_FIXED | PhoneNumberValue::TYPE_VOIP;
+
+$validator = new PhoneNumber([
+    'allowed_types' => $allowed,
+]);
+
+$validator->isValid('+44 (0) 1234 567 890'); // true. GB Fixed Line
+$validator->isValid('+44 (0) 7723 123 456'); // false. GB Mobile
+```
+
+The possible types are as follows:
+
+- `PhoneNumberValue::TYPE_FIXED`
+- `PhoneNumberValue::TYPE_MOBILE`
+- `PhoneNumberValue::TYPE_TOLL_FREE`
+- `PhoneNumberValue::TYPE_PREMIUM_RATE`
+- `PhoneNumberValue::TYPE_SHARED_COST`
+- `PhoneNumberValue::TYPE_VOIP`
+- `PhoneNumberValue::TYPE_PERSONAL`
+- `PhoneNumberValue::TYPE_PAGER`
+- `PhoneNumberValue::TYPE_UAN`
+- `PhoneNumberValue::TYPE_EMERGENCY`
+- `PhoneNumberValue::TYPE_VOICEMAIL`
+- `PhoneNumberValue::TYPE_SHORT_CODE`
+- `PhoneNumberValue::TYPE_STANDARD_RATE`
+- `PhoneNumberValue::TYPE_UNKNOWN`
+
+And there are some additional convenience constants for common use-cases:
+
+- `PhoneNumberValue::TYPE_KNOWN` - All types except `TYPE_UNKNOWN`
+- `PhoneNumberValue::TYPE_ANY` - Any type of valid number
+- `PhoneNumberValue::TYPE_RECOMMENDED` - Fixed lines, mobiles and Voip numbers
+
+## Summary of Constructor Options
+
+```php
+use Laminas\I18n\PhoneNumber\PhoneNumberValue;
+use Laminas\I18n\PhoneNumber\Validator\PhoneNumber;
+
+$options = [
+    'country' => 'ZA', // Can also be a locale string such as 'fr_FR'
+    'country_context' => 'your-country-field-name',
+    'allowed_types' => PhoneNumberValue::TYPE_MOBILE,
+];
+
+$validator = new PhoneNumber($options);
+// Options can also be changed at runtime with:
+$validator->setOptions($options);
+```
+
+## Runtime Modification of Options
+
+Each of the 3 options have companion "setters" to change the runtime behaviour of the validator after it has been constructed:
+
+```php
+use Laminas\I18n\PhoneNumber\PhoneNumberValue;
+use Laminas\I18n\PhoneNumber\Validator\PhoneNumber;
+
+$validator = new PhoneNumber();
+
+$validator->setCountry('US'); // Set the default country to US
+$validator->setCountry('de_DE'); // A locale string can also be used to set the country
+
+$validator->setCountryContext('my-country-input'); // Set the country validation context key
+
+$validator->setAllowedTypes(PhoneNumberValue::TYPE_VOIP); // Set the acceptable number types
+```
```

Comment:
Longer descriptions does not belong in code example. Headlines should be used to break longer text sections to brighten up.

Example:

``````suggestion
### Set Country Code

```php
$validator = new Laminas\I18n\PhoneNumber\Validator\PhoneNumber();
$validator->setCountry('US');
```

A locale string can also be used to set the country.

```php
$validator = new Laminas\I18n\PhoneNumber\Validator\PhoneNumber();
$validator->setCountry('de_DE');
```

### Set the Country Validation Context Key

```php
$validator = new Laminas\I18n\PhoneNumber\Validator\PhoneNumber();
$validator->setCountryContext('my-country-input');
```

### Set the Acceptable Number Types

```php
$validator = new Laminas\I18n\PhoneNumber\Validator\PhoneNumber();
$validator->setAllowedTypes(
    Laminas\I18n\PhoneNumber\PhoneNumberValue::TYPE_VOIP
);
```
``````